### PR TITLE
[ci] Script to automate download & upload wheels to Test PyPI

### PR DESCRIPTION
**Pull request by khluu**
## Changes
- Sample script to download wheels from S3 bucket & upload them to TestPyPI
- The wheels downloaded are stored in a temp directory which gets cleaned up upon exit
- The script uses Test PyPI token which is pending on how we want to store it.

## Checks
- The script downloads expected wheels
- `twine upload` command works on my own repo, but not on ray/ray-cpp since the wheels already exist up there.

### CODE REVIEW

**Comment by khluu** [comment]
@aslonnie  Here is my draft for the script. Once the credential storage gets figured out, I think we can port this in the `release-automation` BK pipeline, like running it as a step/group before running wheel validation group?

File: ci/ray_ci/automation/wheels_pypi.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    exit 1
+fi
+
+TEST_PYPI_TOKEN="test_pypi_token"
+
+download_wheel() {
+    # Download single wheel from given URL into TMP_DIR/<wheel_filename>.whl
+    WHEEL_URL=$1
+    OUTPUT_FILE="${TMP_DIR}/${WHEEL_URL##*/}"
+    if [ "${OVERWRITE-}" == "1" ] || [ ! -f "${OUTPUT_FILE}" ]; then
```
Comment (aslonnie) [inline ci/ray_ci/automation/wheels_pypi.sh:19]: nit: `[[ "${OVERWRITE-}" == "1" || ! -f "${OUTPUT_FILE} ]]`

github copilot learned a lot of bad shell script styling..

File: ci/ray_ci/automation/wheels_pypi.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        wget -O "${OUTPUT_FILE}" "${WHEEL_URL}"
+    fi
+}
+
+download_wheels() {
+    # Iterate through all Python versions, platforms, and ray types and download wheels
+    python_versions=("cp38-cp38" "cp39-cp39" "cp310-cp310" "cp311-cp311")
```
Comment (aslonnie) [inline ci/ray_ci/automation/wheels_pypi.sh:26]: nit: `PYTHON_VERIONS`, use all caps for vars

File: ci/ray_ci/automation/wheels_pypi.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    platforms=("manylinux2014_x86_64" "manylinux2014_aarch64" "macosx_10_15_x86_64" "macosx_11_0_arm64" "win_amd64")
+    ray_types=("ray" "ray_cpp")
+
+    url_prefix="https://ray-wheels.s3-us-west-2.amazonaws.com/releases"
+    for platform in "${platforms[@]}"; do
+        for ray_type in "${ray_types[@]}"; do
+            for python_version in "${python_versions[@]}"; do
```
Comment (aslonnie) [inline ci/ray_ci/automation/wheels_pypi.sh:33]: I feel that we should have a centralized python script to generate this kind of arrays. this is re-defined in so many different places.

File: ci/ray_ci/automation/wheels_pypi.sh (hunk 1 above)
Comment (khluu) [reply ci/ray_ci/automation/wheels_pypi.sh:19]: oh I copied this part straight from `release/util/download_wheels.sh`. Will fix this

**Comment by khluu** [comment]
Closing this PR since I'll be using this script as a step in the `release-automation` pipeline in another PR.

**Pull request closed without merging**